### PR TITLE
feat: `VerifyCommitLight` and `VerifyCommitLightTrusting` _never_ check all signatures

### PR DESCRIPTION
**Pull request by czarcas7ic**
---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: evidence/pool.go [hunk 1]
```diff
@@ -132,19 +132,19 @@ func (evpool *Pool) Update(state sm.State, ev types.EvidenceList) {
 
 // AddEvidence checks the evidence is valid and adds it to the pool.
 func (evpool *Pool) AddEvidence(ev types.Evidence) error {
-	evpool.logger.Debug("Attempting to add evidence", "ev", ev)
+	evpool.logger.Info("Attempting to add evidence", "ev", ev)
```
Comment (czarcas7ic) [inline evidence/pool.go:135]: Do these really need to be info logs? If not im switching them back

File: blocksync/reactor.go [hunk 2]
```diff
@@ -373,7 +373,8 @@ FOR_LOOP:
 			// NOTE: we can probably make this more efficient, but note that calling
 			// first.Hash() doesn't verify the tx contents, so MakePartSet() is
 			// currently necessary.
-			err = state.Validators.VerifyCommitLight(
+			// TODO(sergio): Should we also validate against the extended commit?
+			err = state.Validators.VerifyCommitLightAllSignatures(
```
Comment (ValarDragon) [inline blocksync/reactor.go]: Either we should undo this line, or backport https://github.com/cometbft/cometbft/pull/1858/files as well

File: evidence/pool.go (hunk 1 above)
Comment (ValarDragon) [reply evidence/pool.go:135]: Its fine to add them as Info, evidence is rare

**Review by ValarDragon — APPROVED** [review]
LGTM, but see my comment

File: blocksync/reactor.go (hunk 2 above)
Comment (czarcas7ic) [reply blocksync/reactor.go]: Done here, will merge and backport after tests pass, thank you 

https://github.com/osmosis-labs/cometbft/pull/11/commits/e61ef5f61112b43b0e32b4fd0c5664992a19b672

**Pull request merged**